ReconciliationUI: Display Source Product Gender & Style Changes

vahidbazzaz opened the pull request:
This PR updates the style and text of reconciliation UI, as well as correctly displaying source product gender.

vahidbazzaz commented:
![](https://github.trello.services/images/mini-trello-icon.png) [2.0 Bug: #tiny ReconciliationUI does not display SourceProduct Gender](https://trello.com/c/xrOtJFWC/1320-20-bug-tiny-reconciliationui-does-not-display-sourceproduct-gender)

vahidbazzaz commented:
![](https://github.trello.services/images/mini-trello-icon.png) [2.0 #tiny ReceiveOverlay Design Change for Usability](https://trello.com/c/AJ0dbWtH/1326-20-tiny-receiveoverlay-design-change-for-usability)

HaGuesto commented on front/src/components/BoxReconciliationOverlay/components/MatchProductsForm.tsx:
```diff
@@ -155,14 +155,18 @@ export function MatchProductsForm({
           style={{ color: productId?.value === "" ? "#FF0000" : "#000" }}
         >
           {shipmentDetail?.sourceProduct?.name}{" "}
+          {shipmentDetail?.sourceProduct?.gender
```
Could you handle the none case here @vahidbazzaz ?

![image](https://github.com/boxwise/boxtribute/assets/29065125/db68d5ea-20f7-4070-ba18-3c771fa0f36f)

HaGuesto replied:
@aerinsol

aerinsol replied:
Yes drop the (none) in the no gender case

HaGuesto commented on front/src/components/BoxReconciliationOverlay/components/MatchProductsForm.tsx:
```diff
@@ -155,14 +155,18 @@ export function MatchProductsForm({
     <form onSubmit={handleSubmit(onSubmitMatchProductsForm)}>
       <Flex direction="column" gap="2" alignItems="stretch">
```
Nice to have: increase the gap to 4 and put a Box around the Product Gender part and one around the Size. Looks a bit nicer

HaGuesto reviewed (approved):
looks good. feel free to merge, when you addressed the comments

@vahidbazzaz besides my comment in the code, could you also check why the tests are failing?

vahidbazzaz commented:
> looks good. feel free to merge, when you addressed the comments
> 
> @vahidbazzaz besides my comment in the code, could you also check why the tests are failing?

@HaGuesto It seems only sometimes that tests fail in circleci, locally all tests pass
